src/Testcontainers.Pulsar/PulsarBuilder.cs

```diff
@@ -0,0 +1,118 @@
+
+    public const ushort PulsarBrokerDataPort = 6650;
+
+    public const ushort PulsarWebServicePort = 8080;
+
+    public const string StartupScriptFilePath = "/testcontainers.sh";
+
+    public const string SecretKeyFilePath = "/pulsar/secret.key";
+
+    public const string Username = "test-user";
+
+    private static readonly IReadOnlyDictionary<string, string> AuthenticationEnvVars;
+
+    static PulsarBuilder()
+    {
+        const string authenticationPlugin = "org.apache.pulsar.client.impl.auth.AuthenticationToken";
+
+        var authenticationEnvVars = new Dictionary<string, string>();
+        authenticationEnvVars.Add("authenticateOriginalAuthData", "false");
+        authenticationEnvVars.Add("authenticationEnabled", "true");
+        authenticationEnvVars.Add("authorizationEnabled", "true");
+        authenticationEnvVars.Add("authenticationProviders", "org.apache.pulsar.broker.authentication.AuthenticationProviderToken");
+        authenticationEnvVars.Add("brokerClientAuthenticationPlugin", authenticationPlugin);
+        authenticationEnvVars.Add("CLIENT_PREFIX_authPlugin", authenticationPlugin);
+        authenticationEnvVars.Add("PULSAR_PREFIX_authenticationRefreshCheckSeconds", "5");
+        authenticationEnvVars.Add("PULSAR_PREFIX_tokenSecretKey", "file://" + SecretKeyFilePath);
+        authenticationEnvVars.Add("superUserRoles", Username);
+        AuthenticationEnvVars = new ReadOnlyDictionary<string, string>(authenticationEnvVars);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PulsarBuilder" /> class.
+    /// </summary>
+    public PulsarBuilder()
+        : this(new PulsarConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PulsarBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private PulsarBuilder(PulsarConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override PulsarConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <returns></returns>
+    public PulsarBuilder WithAuthentication()
+    {
+        return Merge(DockerResourceConfiguration, new PulsarConfiguration(authenticationEnabled: true))
+            .WithEnvironment(AuthenticationEnvVars);
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <returns></returns>
+    public PulsarBuilder WithFunctionsWorker(bool functionsWorkerEnabled = true)
+    {
+        // TODO: When enabled we need to adjust the wait strategy.
+        return Merge(DockerResourceConfiguration, new PulsarConfiguration(functionsWorkerEnabled: functionsWorkerEnabled));
+    }
```

Comment:
We will need to adjust the default wait strategy if [functions worker](https://github.com/testcontainers/testcontainers-java/blob/13c2641ff1ff8e74f37594bff96275811a7d809b/modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java#L118-L120) is enabled (we can use a similar approach like we are doing in [Couchbase](https://github.com/testcontainers/testcontainers-dotnet/blob/9156593723eb0c083d9de2741c954806e164061f/src/Testcontainers.Couchbase/CouchbaseBuilder.cs#L81)).

Reply:
Very good feedback. I'll use the same approach as them 👍

Reply:
I have used the approach you suggested. 😄